shared/api/container.go

```diff
@@ -2,18 +2,24 @@ package api
 
 import (
 	"time"
+
+	"github.com/lxc/lxd/shared/instance"
 )
 
-// ContainersPost represents the fields available for a new LXD container
-type ContainersPost struct {
+// InstancesPost represents the fields available for a new LXD instance.
+type InstancesPost struct {
 	ContainerPut `yaml:",inline"`
 
 	Name   string          `json:"name" yaml:"name"`
+	Type   instance.Type   `json:"type" yaml:"type"`
```

Comment:
Needs a comment to indicate it was introduced with the `instances` API extension